gnucash/report/reports/standard/category-barchart.scm

```diff
@@ -600,17 +620,23 @@ developing over time"))
 
                 (define (make-cell contents)
                   (gnc:make-html-table-cell/markup "number-cell" contents))
+                  
+                (define (make-cell-percent contents)
+                  (gnc:make-html-table-cell/markup "number-cell" contents " %"))
 
                 (for-each
                  (lambda (date row)
                    (gnc:html-table-append-row!
                     table
-                    (append (list (make-cell date))
-                            (map make-cell row)
-                            (if cols>1?
-                                (list
-                                 (make-cell (apply gnc:monetary+ row)))
-                                '()))))
+                    (let ((grandt (apply l1norm (map gnc:gnc-monetary-amount row))))
+                     (append (list (make-cell date))
+                             (if ratio-chart? 
+                                 (map (lambda (x) (make-cell-percent (* (safe-/ (gnc:gnc-monetary-amount x) grandt) 100))) row)
+                                 (map make-cell row))
+                             (if cols>1?
+                                 (list
+                                  (make-cell (apply gnc:monetary+ row)))
+                                 '())))))
                  date-string-list
                  (apply zip (map cadr all-data)))
```

Comment:
Instead of recreating grandt for each row, it'll be much more clever to reuse the `grandts` already generated. For that matter, you'll want to modify the `for-each` as follows:

```scheme
(for-each
 (lambda (date row grandt)
  (gnc:html-table-append-row! ...etc...and use grandt now available for use))
 date-string-list
 (apply zip (map cadr all-data))
 grandts)
```